CRM/Event/Import/Parser/Participant.php

```diff
@@ -509,4 +509,167 @@ public function &getImportedParticipations() {
   public function fini() {
   }
 
+  /**
+   * Format values
+   *
+   * @todo lots of tidy up needed here - very old function relocated.
+   *
+   * @param array $values
+   * @param array $params
+   *
+   * @return array|null
+   */
+  protected function formatValues(&$values, $params) {
+    $fields = CRM_Event_DAO_Participant::fields();
+    _civicrm_api3_store_values($fields, $params, $values);
+
+    $customFields = CRM_Core_BAO_CustomField::getFields('Participant', FALSE, FALSE, NULL, NULL, FALSE, FALSE, FALSE);
+
+    foreach ($params as $key => $value) {
+      // ignore empty values or empty arrays etc
+      if (CRM_Utils_System::isNull($value)) {
+        continue;
+      }
+
+      // Handling Custom Data
+      if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($key)) {
+        $values[$key] = $value;
+        $type = $customFields[$customFieldID]['html_type'];
+        if ($type == 'CheckBox' || $type == 'Multi-Select') {
+          $mulValues = explode(',', $value);
+          $customOption = CRM_Core_BAO_CustomOption::getCustomOption($customFieldID, TRUE);
+          $values[$key] = array();
+          foreach ($mulValues as $v1) {
+            foreach ($customOption as $customValueID => $customLabel) {
+              $customValue = $customLabel['value'];
+              if ((strtolower(trim($customLabel['label'])) == strtolower(trim($v1))) ||
+                (strtolower(trim($customValue)) == strtolower(trim($v1)))
+              ) {
+                if ($type == 'CheckBox') {
+                  $values[$key][$customValue] = 1;
+                }
+                else {
+                  $values[$key][] = $customValue;
+                }
+              }
+            }
+          }
+        }
+        elseif ($type == 'Select' || $type == 'Radio') {
+          $customOption = CRM_Core_BAO_CustomOption::getCustomOption($customFieldID, TRUE);
+          foreach ($customOption as $customFldID => $customValue) {
+            $val = CRM_Utils_Array::value('value', $customValue);
+            $label = CRM_Utils_Array::value('label', $customValue);
+            $label = strtolower($label);
+            $value = strtolower(trim($value));
+            if (($value == $label) || ($value == strtolower($val))) {
+              $values[$key] = $val;
+            }
+          }
+        }
+      }
+
+      switch ($key) {
+        case 'participant_contact_id':
+          if (!CRM_Utils_Rule::integer($value)) {
+            return civicrm_api3_create_error("contact_id not valid: $value");
+          }
+          $dao = new CRM_Core_DAO();
+          $qParams = array();
+          $svq = $dao->singleValueQuery("SELECT id FROM civicrm_contact WHERE id = $value",
```

Comment:
I've tidied those up My real focus is on the contact import - https://github.com/civicrm/civicrm-core/pull/10579 (although I've tried to keep that PR as simple as possible)